Source/Artifacts/Artifact.cs

```diff
@@ -9,8 +9,18 @@ namespace Dolittle.SDK.Artifacts
     /// <summary>
     /// Represents the concept of an artifact.
     /// </summary>
-    public abstract class Artifact : Value<Artifact>
+    public class Artifact : Value<Artifact>
```

Comment:
Not sure I understand why it needs to be !abstract. We shouldn't really ever be instantiating it, should we?

Reply:
If we do it like we do in the JS SDK this needs to be !abstract. Unless we only have methods that takes in Artifact and not ArtifactId. It makes more sense to have it !abstract if we go the route where we extend the Artifact class with EventType, etc...

We can go the route of having it abstract, but then we need to have an EventType class extending Artifact and have a builder thing for EventType that can take in an ArtifactId and register it to the Artifacts